packages/bson-transpilers/symbols/php/templates.yaml

```diff
@@ -0,0 +1,1271 @@
+            if (arg.charAt(0) === '"' && arg.charAt(arg.length - 1) === '"') {
+                return `(${arg})`;
+            }
+            return `('${arg}')`;
+        }
+    MinKeySymbolTemplate: &MinKeySymbolTemplate !!js/function >
+        () => {
+            return 'new \\MongoDB\\BSON\\MinKey';
+        }
+    MinKeySymbolArgsTemplate: &MinKeySymbolArgsTemplate !!js/function >
+        () => {
+            return `()`;
+        }
+    MaxKeySymbolTemplate: &MaxKeySymbolTemplate !!js/function >
+        () => {
+            return 'new \\MongoDB\\BSON\\MaxKey';
+        }
+    MaxKeySymbolArgsTemplate: &MaxKeySymbolArgsTemplate !!js/function >
+        () => {
+            return `()`;
+        }
+    TimestampSymbolTemplate: &TimestampSymbolTemplate !!js/function >
+        () => {
+            return 'new \\MongoDB\\BSON\\Timestamp';
+        }
+    TimestampSymbolArgsTemplate: &TimestampSymbolArgsTemplate !!js/function >
+        (lhs, arg1, arg2) => {
+            return `(${arg2 === undefined ? 0 : arg2}, ${arg1 === undefined ? 0 : arg1})`;
+        }
+    # non bson-specific
+    NumberSymbolTemplate: &NumberSymbolTemplate !!js/function >
+        () => ''
+    NumberSymbolArgsTemplate: &NumberSymbolArgsTemplate !!js/function >
+        (lhs, arg, type) => {
+            arg = arg === undefined ? 0 : arg;
+
+            switch(type) {
+                case '_string':
+                    if (arg.indexOf('.') !== -1) {
+                        return `(float) ${arg}`
+                    }
+                    return `(int) ${arg}`
+                default:
+                    return `${arg}`
+            }
+        }
+    DateSymbolTemplate: &DateSymbolTemplate !!js/function >
+        () => {
+            return '\\DateTime';
+        }
+    DateSymbolArgsTemplate: &DateSymbolArgsTemplate !!js/function >
+        (lhs, date, isString) => {
+            if (date === null) {
+                return `new ${lhs}('now', new \\DateTimeZone('UTC'))`;
+            }
+        
+            const dateData = [
+                date.getUTCFullYear(),
+                (date.getUTCMonth() + 1).toString().length == 1 ? ['0', date.getUTCMonth() + 1].join('') : date.getUTCMonth() + 1,
+                (date.getUTCDate()).toString().length == 1 ? ['0', date.getUTCDate()].join('') : date.getUTCDate()
+            ].join('-');
+        
+            const timeData = [
+                (date.getUTCHours()).toString().length == 1 ? ['0', date.getUTCHours()].join('') : date.getUTCHours(),
+                (date.getUTCMinutes()).toString().length == 1 ? ['0', date.getUTCMinutes()].join('') : date.getUTCMinutes(),
+                (date.getUTCSeconds()).toString().length == 1 ? ['0', date.getUTCSeconds()].join('') : date.getUTCSeconds()
+            ].join(':');
+        
+        
+            const dateStr = [
+                dateData,
+                ' ',
+                timeData
+            ].join('');
+
+            return isString 
```

Comment:
When changing to use UTCDateTime, I expect you'll have a simpler code path when `isString` is false, since you can use the integer as-is. I'm not sure what the unit is for an integer parameter, but if it's seconds you'll need to multiply accordingly to provide UTCDateTime with milliseconds.

OK to resolve.

Reply:
Here is the new template:
````
        (lhs, date, isString) => {
            if (date === null) {
                return `new ${lhs}()`;
            }
            return isString 
                ? `(new ${lhs}(${date.getTime()}))->toDateTime()->format(\\DateTimeInterface::RFC3339_EXTENDED)`
                : `new ${lhs}(${date.getTime()})`
            ;
        }
````
Function getTime() returns time in milliseconds.